core/utils/endian.h

```diff
@@ -152,6 +153,15 @@ class[[gnu::packed]] BigEndian final : public EndianBase<T> {
     return os;
   }
 
+  const std::vector<uint8_t> ToByteVector() const {
+    union {
+      T data;
+      uint8_t bytes[0];
```

Comment:
Why `0` instead of `sizeof(T)`? (Should be OK either way, just seems a bit odd.)

Reply:
Oh thanks. I was fiddling with non-template version of this but forgot go rollback. Fixed.